### DISCUSSION TITLE
Obtaining the objective spaces values per generation

### DISCUSSION THREAD

**Discussion by RuiRibeiro73** [anyoptimization/pymoo#246] [Q&A] [answered]
Hello first of all thank you for your hard work!.
I aiming to  solve a multi-objective optimization problem using NSGA II and  i need to see the objective spaces values per generation so that i can plot the evolution of the Pareto Curve per generation.  Basically i want to plot multiple Pareto curves one for each 5 generations.  I cannot use save history because my optimization uses AutoML to predict certain values that necessary in my problem. Is there a way to obtain the objective spaces values per generation of the non-dominated solutions?
Best regards
Rui

**Comment by blankjul** [comment]
Yes, check out the `Callback` tutorial. This is exactly what you are trying to do.
https://pymoo.org/interface/callback.html

**Reply by RuiRibeiro73** [reply]
Thank you for your fast reply but it is not really what i need. Because it gives me all the solutions ( in my case since i have a pop size = 100 i get 100 solutions) . I  did not explained corretly but ill will try with an small example.
I have 2 generations of 100 individues. In the first generation i have 3 non dominated solutions. In the second 6.  Is possible to get for the first generation 3 objective spaces values and for the second generation 6 objective spaces values since they are the nds.  I implemented your callback using your example  removing min because it only returns one value.
Thank you
Rui

**Comment by blankjul** [comment, marked as answer]
You can use 
```
algorithm.opt.get("F")
```

to obtain the objective space values of the current optimum.
(this is just an example in that tutorial I have sent you)

**Reply by RuiRibeiro73** [reply]
Thank you very much!!
